File: Army/berserk.cpp
#include "berserk.h"

using namespace std;

Berserk::Berserk(const string& name, int healthPoint, int damage) : Unit(name, healthPoint, damage) {}

Berserk::~Berserk() {}

void Berserk::takeMagicDamage(int damage) {
    
}

File: Army-Java/Units/Berserk.java
package com.gymfox.Army.Units;

import com.gymfox.Army.Ability.BerserkAbility;

public class Berserk extends Unit {
    public Berserk(String name, int healthPointLimit, int damage) {
        super(name, healthPointLimit, damage);
        this.ability = new BerserkAbility(this);
        this.immunityToMagic = true;
    }
}
